### ISSUE TITLE
Python 3.10: not all pre-commit hooks supported

### ISSUE THREAD

**Issue by kreczko** [scikit-hep/cookie#52]
While #41 introduced Python 3.10 to the CI, not all pre-commit hooks support that version.

One such hook is `pycl`:

```bash
ERROR: Package 'pycln' requires a different Python: 3.10.0 not in '<3.10,>=3.6.2'
```

It is still an open issue: https://github.com/hadialqattan/pycln/issues/78

In addition, even the `pyproject.toml` templates include lines like
```toml
# https://github.com/scikit-hep/cookie/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/pyproject-flit.toml#L34
requires = [
  "typing_extensions>=3.7; python_version<'3.8'",
]
```
So I assume that this will also create failures.

**Comment by hadialqattan** [comment]
Hi, I'm the creator of [Pycln ](https://github.com/hadialqattan/pycln). I feel sorry about being too late, but I want to tell you that Python3.10 support has been added to Pycln ([v1.1.0](https://github.com/hadialqattan/pycln/releases/tag/v1.1.0)).

**Comment by kreczko** [comment]
Can be closed once #54 is merged.
